hazelcast/src/main/java/com/hazelcast/multimap/impl/operations/MergeOperation.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.multimap.impl.operations;
+
+import com.hazelcast.core.EntryEventType;
+import com.hazelcast.multimap.impl.MultiMapContainer;
+import com.hazelcast.multimap.impl.MultiMapDataSerializerHook;
+import com.hazelcast.multimap.impl.MultiMapMergeContainer;
+import com.hazelcast.multimap.impl.MultiMapRecord;
+import com.hazelcast.multimap.impl.MultiMapValue;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.spi.BackupAwareOperation;
+import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.SplitBrainMergeEntryView;
+import com.hazelcast.spi.SplitBrainMergePolicy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static com.hazelcast.spi.merge.SplitBrainEntryViews.createSplitBrainMergeEntryView;
+import static com.hazelcast.util.MapUtil.createHashMap;
+
+public class MergeOperation extends MultiMapOperation implements BackupAwareOperation {
+
+    private List<MultiMapMergeContainer> mergeEntries;
+    private SplitBrainMergePolicy mergePolicy;
+
+    private transient Map<Data, Collection<MultiMapRecord>> resultMap;
+
+    public MergeOperation() {
+    }
+
+    public MergeOperation(String name, List<MultiMapMergeContainer> mergeEntries,
+                          SplitBrainMergePolicy mergePolicy) {
+        super(name);
+        this.mergeEntries = mergeEntries;
+        this.mergePolicy = mergePolicy;
+    }
+
+    @Override
+    public void run() throws Exception {
+        MultiMapContainer container = getOrCreateContainer();
+        resultMap = createHashMap(mergeEntries.size());
+        for (MultiMapMergeContainer mergeEntry : mergeEntries) {
+            Data key = mergeEntry.getKey();
+            if (!container.canAcquireLock(key, getCallerUuid(), -1)) {
```

Comment:
Since there is no built-in support for multiple keys in `BlockingOperation` I did the simple solution of just skipping locked keys in the merge process.